WalletWasabi/BitcoinCore/Configuration/CoreConfigTranslator.cs

```diff
@@ -0,0 +1,78 @@
+using NBitcoin;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using WalletWasabi.BitcoinCore.Configuration.Whitening;
+using WalletWasabi.Helpers;
+
+namespace WalletWasabi.BitcoinCore.Configuration
+{
+	public class CoreConfigTranslator
+	{
+		public CoreConfig Config { get; }
+		public Network Network { get; }
+
+		public CoreConfigTranslator(CoreConfig config, Network network)
+		{
+			Config = Guard.NotNull(nameof(config), config);
+			Network = Guard.NotNull(nameof(network), network);
+		}
+
+		public string TryGetValue(string key)
+		{
+			var configDic = Config.ToDictionary();
+			string result = null;
+			foreach (var networkPrefixWithDot in NetworkTranslator.GetConfigPrefixesWithDots(Network))
+			{
+				var found = configDic.TryGet($"{networkPrefixWithDot}{key}");
+				if (found is { })
+				{
+					result = found;
```

Comment:
Shouldn't return/break here?

Reply:
No. I tested how Bitcoin Core behaves, and it always considers the last entry to be the valid one. Eg if you have:

```
testnet = 1
testnet = 0
```

then it will not throw exception, but consider `testnet = 0` as the valid entry.